### PR TITLE
Updates for Cataclysm Classic

### PR DESCRIPTION
**Pull request by Stanzilla**
We can probably remove Wrath if they do the same as with TBC. I also added the new TOC load directive (`AllowLoadGameType`) to the Cata TOC only for now. If it ends up working how it is supposed to, we can maybe reduce our version checking code.

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Comment by Merfinx** [comment]
I'm most likely dumb and posting it in wrong place, but could we have "Talent specialization" in Load Tab, like we do on Retail. There is an API function that returns the values 

`
GetPrimaryTalentTree()
--primaryTabId = GetPrimaryTalentTree([isInspect[, isPet[, talentGroup]]])
`

I don't know why this func exists because it was removed in MoP, but yeah

File: WeakAuras/WeakAuras_Cata.toc [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+## Globe-Post: WeakAurasOptions, WeakAurasModelPaths, WeakAurasTemplates
+## X-Website: https://www.curseforge.com/wow/addons/weakauras
+## X-Curse-Project-ID: 65387
+## X-WoWI-ID: 24910
+## X-Wago-ID: VBNBxKx5
+## DefaultState: Enabled
+## LoadOnDemand: 0
+## SavedVariables: WeakAurasSaved
+## OptionalDeps: Ace3, LibCompress, LibSharedMedia-3.0, AceGUI-3.0-SharedMediaWidgets, Masque, GTFO, LibButtonGlow-1.0, LibSpellRange-1.0, LibRangeCheck-3.0, LibDBIcon-1.0, LibGetFrame-1.0, !LibUIDropDownMenu, !!AddonLocale, BigWigs, DBM-Core
+## AllowLoadGameType: cata
```
Comment (Stanzilla) [inline WeakAuras/WeakAuras_Cata.toc:21]: Meo said these don't work for standard addons but doesn't hurt to keep them in I guess

File: WeakAurasOptions/RegionOptions/Model.lua [hunk 2]
```diff
@@ -253,7 +253,7 @@ local function createThumbnail()
 
   local model = CreateFrame("PlayerModel", nil, borderframe);
   borderframe.model = model;
-  model.SetTransformFixed = model.GetResizeBounds and ModelSetTransformFixed or model.SetTransform -- TODO change test to WeakAuras.IsWrathOrRetail() after 3.4.1 release
+  model.SetTransformFixed = model.GetResizeBounds and ModelSetTransformFixed or model.SetTransform -- TODO change test to WeakAuras.IsWrathOrCataOrRetail() after 3.4.1 release
```
Comment (InfusOnWoW) [inline WeakAurasOptions/RegionOptions/Model.lua:256]: This doesn't look right.

File: WeakAuras/Prototypes.lua [hunk 3]
```diff
@@ -9498,7 +9500,7 @@ Private.event_prototypes = {
         type = "number",
         init = "select(2, UnitResistance('player', 3))",
```
Comment (InfusOnWoW) [inline WeakAuras/Prototypes.lua:9501]: does UnitResistance work on cata?

File: WeakAuras/Prototypes.lua [hunk 4]
```diff
@@ -2200,7 +2200,7 @@ Private.event_prototypes = {
         store = true,
         conditionType = "select",
         enable = function(trigger)
-          return WeakAuras.IsWrathOrRetail() and (trigger.unit == "group" or trigger.unit == "raid" or trigger.unit == "party")
+          return WeakAuras.IsWrathOrCataOrRetail() and (trigger.unit == "group" or trigger.unit == "raid" or trigger.unit == "party")
         end
       },
       {
```
Comment (InfusOnWoW) [inline WeakAuras/Prototypes.lua:2206]: I think Raid Role below should be for catat too.

File: WeakAuras/Prototypes.lua (hunk 3 above)
Comment (mrbuds) [reply WeakAuras/Prototypes.lua:9501]: On current build it does

File: WeakAuras/Types.lua [hunk 5]
```diff
@@ -1490,7 +1490,7 @@ if WeakAuras.IsRetail() then
   Private.GetCurrencyIDFromLink = C_CurrencyInfo.GetCurrencyIDFromLink
   Private.ExpandCurrencyList = C_CurrencyInfo.ExpandCurrencyList
   Private.GetCurrencyListInfo = C_CurrencyInfo.GetCurrencyListInfo
-elseif WeakAuras.IsWrathClassic() then
+elseif WeakAuras.IsWrathOrCata() then
```
Comment (InfusOnWoW) [inline WeakAuras/Types.lua:1493]: WeakAuras.race_types needs to be checked, cata added worgen and goblin

Comment (mrbuds) [reply WeakAuras/Types.lua:1493]: They are on load options

File: WeakAuras/GenericTrigger.lua [hunk 6]
```diff
@@ -2366,7 +2366,7 @@ do
     cdReadyFrame:RegisterEvent("SPELLS_CHANGED");
```
Comment (InfusOnWoW) [inline WeakAuras/GenericTrigger.lua:2366]: Do we need TenchInit for temporary enchants?

File: WeakAuras/Types.lua [hunk 7]
```diff
@@ -3944,7 +3951,7 @@ for i = 1, 4 do
   Private.multiUnitUnits.party["partypet"..i] = true
 end
 
-if WeakAuras.IsWrathOrRetail() then
+if WeakAuras.IsWrathOrCataOrRetail() then
```
Comment (InfusOnWoW) [inline WeakAuras/Types.lua:3954]: There's skippedWeaponTypes below which probably need adjustment too.

File: WeakAuras/BuffTrigger2.lua [hunk 8]
```diff
@@ -3129,7 +3129,7 @@ function BuffTrigger.Add(data)
 
       local groupTrigger = trigger.unit == "group" or trigger.unit == "raid" or trigger.unit == "party"
       local effectiveIgnoreSelf = (groupTrigger or trigger.unit == "nameplate") and trigger.ignoreSelf
-      local effectiveGroupRole = WeakAuras.IsWrathOrRetail() and (groupTrigger and trigger.useGroupRole and trigger.group_role) or nil
+      local effectiveGroupRole = WeakAuras.IsWrathOrCataOrRetail() and (groupTrigger and trigger.useGroupRole and trigger.group_role) or nil
```
Comment (InfusOnWoW) [inline WeakAuras/BuffTrigger2.lua:3132]: useGroupRole in the setings is set to IsClassicEra, that needs adjustments.

File: WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasMiniTalent_Cata.lua [hunk 9]
```diff
@@ -0,0 +1,327 @@
+if not WeakAuras.IsLibsOK() then
```
Comment (InfusOnWoW) [inline WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasMiniTalent_Cata.lua:1]: SpinBox needs a look too, probably same as wrath/classic.

File: WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasMiniTalent_Cata.lua [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+  return
```
Comment (InfusOnWoW) [inline WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasMiniTalent_Cata.lua:2]: Also minimizebutton:SetPoint("RIGHT", frame.CloseButton, "LEFT", WeakAuras.IsClassicEraOrWrath() and  10 or 0, 0) in OptionsFrame.lua

File: WeakAuras/Prototypes.lua [hunk 11]
```diff
@@ -9210,9 +9212,9 @@ Private.event_prototypes = {
         type = "number",
```
Comment (InfusOnWoW) [inline WeakAuras/Prototypes.lua:9212]: Spirit still exists

**Comment by InfusOnWoW** [comment]
I did a few spot checks, but we probably need to look at each IsXXX call to see whether they are correct or not. That's around 700 places, so it's actually not that bad.

I'll do that once I actually have beta and when I can motivate myself to do that.

**Comment by mrbuds** [comment]
> I'm most likely dumb and posting it in a wrong place, and i'm sorry for that in advance, but could we have "Talent specialization" in Load Tab, like we do on Retail, please. There is an API function that returns the values
> 
> `GetPrimaryTalentTree() --primaryTabId = GetPrimaryTalentTree([isInspect[, isPet[, talentGroup]]])`
> 
> I don't know why this func exists because it was removed in MoP, but yeah

GetTalentTabInfo return the specID which make it possible for a spec filter for all classes
But right now all API that would allow to enumerate specs and get their localized name are not implemented, despite them used in some blizzard addons that are disabled like Blizzard_Wardrobe

I'll check that in a later build

File: WeakAuras/BuffTrigger2.lua (hunk 8 above)
Comment (mrbuds) [reply WeakAuras/BuffTrigger2.lua:3132]: It's for the "hidden" property, it's correct but edgy

File: WeakAuras/Types.lua (hunk 5 above)
Comment (mrbuds) [reply WeakAuras/Types.lua:1493]: it's good 
image

File: WeakAuras/Prototypes.lua (hunk 3 above)
Comment (mrbuds) [reply WeakAuras/Prototypes.lua:9501]: It works, but it may be one of the things blizzard forgot to remove and will be drop later

**Comment by seathasky** [comment]
**Cata beta Bug:**
Color picker does not save picked color.
Included is video showcasing this.

https://github.com/WeakAuras/WeakAuras2/assets/17598024/b2f2fb43-35fc-4b49-ae2d-6da0bc446219

File: WeakAurasOptions/OptionsFrames/ModelPicker.lua [hunk 12]
```diff
@@ -220,7 +220,7 @@ local function ConstructModelPicker(frame)
   group:AddChild(modelTree);
 
   local model = CreateFrame("PlayerModel", nil, group.content);
-  model.SetTransformFixed = model.GetResizeBounds and OptionsPrivate.Private.ModelSetTransformFixed or model.SetTransform -- TODO change test to WeakAuras.IsWrathOrRetail() after 3.4.1 release
+  model.SetTransformFixed = model.GetResizeBounds and OptionsPrivate.Private.ModelSetTransformFixed or model.SetTransform -- TODO change test to WeakAuras.IsWrathOrCataOrRetail() after 3.4.1 release
```
Comment (InfusOnWoW) [inline WeakAurasOptions/OptionsFrames/ModelPicker.lua:223]: @mrbuds what was the context here?

**Comment by InfusOnWoW** [comment]
I also think merging early is probably a good idea, as is I believe this can't break anything for live/wrath.

File: WeakAurasOptions/OptionsFrames/ModelPicker.lua (hunk 12 above)
Comment (mrbuds) [reply WeakAurasOptions/OptionsFrames/ModelPicker.lua:223]: This was changed when new model API from Retail was retrofitted to wotlk in the middle of an extension ("classic_ptr" had different api than "classic")

This test can be remove now, all versions of the game including sod are using the same API

**Comment by mushymeterreader** [comment]
Hey, I'm trying to build a boomy eclipse bar and I noticed in Trigger->Player/Unit Info->Power->Power type that "eclipse" is missing. I cross referenced with a private server version of WA. There is a bunch of retail ones in the cata beta one.

**Comment by Kirschie** [comment]
Resolved this working with the team in Discord but reporting here for tracking

This is what I was experiencing with with build 'WeakAuras-PR#4904-d7a25c4249fbb968436483e39d544700e3a0d700.zip'

- When I use /wa nothing opens up
- In the minmap I can see WA is loaded, and in ElvUi it shows WA is loaded into memory (57mb~)
- When I left click the minmap WA button, WA does not open
- I can right click it and track events
- I do not see any settings for WA under Options > Addons > WA is not listed
- This is what I see in Bug Sack for WA when loading into the game, no errors presented when running /wa or left clicking https://pastebin.com/kygZQ40L

```
1x Usage: local result = C_Item.IsEquippedItem(itemInfo)
[string "=[C]"]: in function `IsEquippedItem'
[string "return function(event, combat, alive, encounter, vehicle, vehicleUi, class, player, realm, race, faction, level, role, raid_role, ingroup, groupSize, group_leader, zone, zoneId, zonegroupId, instanceId, minimapZoneText, encounterid, size, difficulty, instance_type)"]:2: in function `loadFunc'
```

Root Cause / Resolution
- I copied over the saved variables from another character from WOTLK to save my WA settings, etc...
- Removing all WA related .lua from WTF\Account\<Account #>\SavedVariables\ resolved my problem
- thank  you to Spaten for the help

**Review by mrbuds — APPROVED** [review]

**Comment by adhdk** [comment]
> > I'm most likely dumb and posting it in a wrong place, and i'm sorry for that in advance, but could we have "Talent specialization" in Load Tab, like we do on Retail, please. There is an API function that returns the values
> > `GetPrimaryTalentTree() --primaryTabId = GetPrimaryTalentTree([isInspect[, isPet[, talentGroup]]])`
> > I don't know why this func exists because it was removed in MoP, but yeah
> 
> GetTalentTabInfo return the specID which make it possible for a spec filter for all classes But right now all API that would allow to enumerate specs and get their localized name are not implemented, despite them used in some blizzard addons that are disabled like Blizzard_Wardrobe
> 
> I'll check that in a later build

i think the APIs you're talking about were included in one of the recent updates, but in addition to the specialization load option it would be great to get the "Class and Specialization" trigger that exists on retail as well

**Comment by mrbuds** [comment]
> i think the APIs you're talking about were included in one of the recent updates, but in addition to the specialization load option it would be great to get the "Class and Specialization" trigger that exists on retail as well

Correct place would be at https://github.com/WeakAuras/WeakAuras2/issues/4919
Could you add more details on what was added? How can you list all possible specIDs? How do you get you current specID?

**Pull request merged**